internal/experiment/generation/stormforgeperfaz.go

```diff
@@ -0,0 +1,203 @@
+/*
+Copyright 2021 GramLabs, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generation
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"unicode"
+
+	"github.com/thestormforge/optimize-go/pkg/config"
+)
+
+// DefaultOptimizeConfig is the mega-hack-sky-hook used to inject configuration
+// this low into the code. This should not be necessary, as generation should not
+// be accessing the configuration directly.
+var DefaultOptimizeConfig *config.OptimizeConfig
+
+// StormForgePerformanceAuthorization is a helper for obtaining StormForge Performance
+// access tokens (JWTs). Normally, low level code like this does not have access to
+// the application layer configuration so this is a bit of a hack to ensure we can
+// talk to the appropriate APIs.
+type StormForgePerformanceAuthorization struct {
+	// The configuration to get authorization information from, leave nil to get the default.
+	Config *config.OptimizeConfig
+	// The function used to lookup environment variables, leave nil to get the default.
+	LookupEnvFunc func(string) (string, bool)
+	// The label to use to describe the Performance Service Account (if needed)
+	ServiceAccountLabel func() string
+}
+
+// AccessToken returns a StormForge Performance access token for the optional organization.
+// If the organization is empty, we use the private space that depends on shared authorization
+// between the two products.
+func (az *StormForgePerformanceAuthorization) AccessToken(ctx context.Context, org string) (string, error) {
+	if err := az.setDefaults(); err != nil {
+		return "", err
+	}
+
+	if org != "" {
+		return az.accessTokenForOrganization(ctx, org)
+	}
+
+	return az.accessTokenForOptimize(ctx)
+}
+
+func (az *StormForgePerformanceAuthorization) accessTokenForOrganization(ctx context.Context, org string) (string, error) {
+	// Check for an organization specific environment variable (this will never conflict with "real forge")
+	key := fmt.Sprintf("STORMFORGER_%s_JWT", strings.Map(toEnv, org))
+	if at, ok := az.lookupEnv(key); ok {
+		return at, nil
+	}
+
+	// Register a new service account
+	sa, err := exec.CommandContext(ctx, "forge", "serviceaccount", "create", org, az.ServiceAccountLabel()).Output()
```

Comment:
One thing to note is this requires the user to be a manager/owner of the current org in forge. Im not sure what the error message looks like if don't have those permissions, but it may be worth enriching the error message to mention this.